internal/middleware/middleware_test.go

```diff
@@ -0,0 +1,583 @@
+				}
+			},
+		},
+		{
+			name: "Failure::ExtractUser:: user id not string",
+			setupFunc: func() (*http.Request, authentication.JWTService) {
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				req.AddCookie(&http.Cookie{
+					Name:  "token",
+					Value: "123",
+				})
+				mockJwtSvc := mock.NewMockJWTService(mockCtrl)
+				token := jwtGo.Token{
+					Claims: jwtGo.MapClaims{"user_id": 1200},
+					Valid:  true,
+				}
+				mockJwtSvc.EXPECT().ValidateToken(gomock.Any()).Return(&token, nil)
+
+				return req, mockJwtSvc
+			},
+			validator: func(res *httptest.ResponseRecorder, hit *bool) {
+				if *hit != false {
+					t.Errorf("Want: %v, Got: %v", false, hit)
+				}
+				by, _ := ioutil.ReadAll(res.Body)
+				result := model.Response{}
+				err := json.Unmarshal(by, &result)
+				if err != nil {
+					t.Log(err)
+					t.Fail()
+					return
+				}
+				expected := &model.Response{
+					Status:  http.StatusBadRequest,
+					Message: codes.GetErr(codes.ErrAssertUserid),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(&result, expected) {
+					t.Errorf("Want: %v, Got: %v", expected, result)
+				}
+			},
+		},
+	}
+
+	// to execute the tests in the table
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// STEP 1: seting up all instances for the specific test case
+			res := httptest.NewRecorder()
+			req, jwt := tt.setupFunc()
+
+			// STEP 2: call the test function
+			middleware := NewTransactionMgmtMiddleware(&config.SvcConfig{
+				JwtSvc: config.JWTSvc{
+					JwtSvc: jwt,
+				},
+				Cfg: &config.Config{},
+			})
+			var hit bool
+			testFunc := test(&hit)
+			x := middleware.ExtractUser(testFunc)
+			x.ServeHTTP(res, req)
+
+			tt.validator(res, &hit)
+
+		})
+	}
+}
+func TestTransactionMgmtMiddleware_Cacher(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tests := []struct {
+		name           string
+		config         config.Config
+		extractMsgFunc func(closer io.ReadCloser) (string, error)
+		setupFunc      func() (*http.Request, *redisMock.MockCacher)
+		validator      func(*httptest.ResponseRecorder, *bool)
+	}{
+		{
+			name:   "SUCCESS::Cacher::Cached Response",
+			config: config.Config{},
+			setupFunc: func() (*http.Request, *redisMock.MockCacher) {
+
+				req := httptest.NewRequest(http.MethodGet, "http://localhost:80", nil)
+				ctx := session.SetSession(req.Context(), model2.SessionStruct{UserId: "123"})
+				mockCacher := redisMock.NewMockCacher(mockCtrl)
+				cacheResponse := model2.CacheResponse{Status: http.StatusOK, Response: "ok", ContentType: "application/json"}
+				b, _ := json.Marshal(cacheResponse)
+				mockCacher.EXPECT().Get("http://localhost:80/auth/123").Return(b, nil)
+				return req.WithContext(ctx), mockCacher
+			},
+			extractMsgFunc: func(closer io.ReadCloser) (string, error) {
+				return "", nil
+			},
+			validator: func(res *httptest.ResponseRecorder, hit *bool) {
+				if *hit != false {
```

Comment:
Validate the headers that are set when cache get is successful.

Reply:
done

Reply:
Are all the headers validated?

Reply:
done